src/utils/redis.js

```diff
@@ -3,14 +3,24 @@ const redis = require('redis');
 const { Configuration } = require('@hpi-schul-cloud/commons');
 
 const { GeneralError } = require('../errors');
+const logger = require('../logger');
 
 let redisClient = false;
 
-function initializeRedisClient() {
+async function initializeRedisClient() {
 	if (Configuration.has('REDIS_URI')) {
 		try {
 			redisClient = redis.createClient({
 				url: Configuration.get('REDIS_URI'),
+				// Legacy mode is needed for compatibility with v4, see https://github.com/redis/node-redis/blob/HEAD/docs/v3-to-v4.md#legacy-mode
+				legacyMode: true,
+			});
+			await redisClient.connect();
+
+			// The error event must be handled, otherwise the app crashes on redis connection errors.
+			// This is due to basic NodeJS behavior: https://nodejs.org/api/events.html#error-events
+			redisClient.on('error', (err) => {
+				logger.error('Redis client error', err);
```

Comment:
Put this bevor the connect, it's better if there is an issue at the connect before the server don't come right up